grails-app/services/org/pih/warehouse/data/ProductAssociationDataService.groovy

```diff
@@ -28,8 +29,32 @@ class ProductAssociationDataService {
             productAssociationInstance.product = Product.findByProductCode(params['product.productCode'])
             productAssociationInstance.associatedProduct = Product.findByProductCode(params['associatedProduct.productCode'])
             params['product.id'] =  productAssociationInstance.product?.id
+            params['product.name'] =  productAssociationInstance.product?.name
             params['associatedProduct.id'] =  productAssociationInstance.associatedProduct?.id
+            params['associatedProduct.name'] =  productAssociationInstance.associatedProduct?.name
 
+            params.hasMutualAssociation = Boolean.valueOf(params.hasMutualAssociation as String) || (params.hasMutualAssociation as String)?.equalsIgnoreCase("yes")
+
+            // check for duplicate association entries in the file
+            def indexOfDuplicate = listOfValidatedProductAssociationParams.findIndexOf{it ->
+                if (params.hasMutualAssociation || it.hasMutualAssociation) {
+                    return  (
```

Comment:
This entire `or` looks weird are you sure it is proper?

Reply:
It works properly but I agree it looks ugly.
the problem I had here was that I had to detect not only duplicates but also a case where we have two-way and one-way entries of the same association
```
Association_1(product: A, associatedProduct: B, type: Type1, hasMutualAssociation: false )
Association_2(product: A, associatedProduct: B, type: Type1, hasMutualAssociation: true )
```
```
Association_1(product: A, associatedProduct: B, type: Type1, hasMutualAssociation: false )
Association_2(product: B, associatedProduct: A, type: Type1, hasMutualAssociation: true )
```